packages/playwright-core/src/server/utils/network.ts

```diff
@@ -109,6 +115,47 @@ export function fetchData(params: HTTPRequestParams, onError?: (params: HTTPRequ
   });
 }
 
+function shouldBypassProxy(url: URL, bypass?: string): boolean {
+  if (!bypass)
+    return false;
+  const domains = bypass.split(',').map(s => {
+    s = s.trim();
+    if (!s.startsWith('.'))
+      s = '.' + s;
+    return s;
+  });
+  const domain = '.' + url.hostname;
+  return domains.some(d => domain.endsWith(d));
+}
+
+export function createProxyAgent(proxy?: ProxySettings, forUrl?: URL) {
+  if (!proxy)
+    return;
+  if (forUrl && proxy.bypass && shouldBypassProxy(forUrl, proxy.bypass))
+    return;
+
+  // Browsers allow to specify proxy without a protocol, defaulting to http.
+  let proxyServer = proxy.server.trim();
+  if (!/^\w+:\/\//.test(proxyServer))
+    proxyServer = 'http://' + proxyServer;
+
+  const proxyOpts = url.parse(proxyServer);
+  if (proxyOpts.protocol?.startsWith('socks')) {
+    return new SocksProxyAgent({
+      host: proxyOpts.hostname,
+      port: proxyOpts.port || undefined,
+    });
+  }
+  if (proxy.username)
+    proxyOpts.auth = `${proxy.username}:${proxy.password || ''}`;
+  if (forUrl && ['ws:', 'wss:'].includes(forUrl.protocol)) {
+    // Force CONNECT method for WebSockets.
```

Comment:
```suggestion
  // Force CONNECT method for WebSockets.
  if (forUrl && ['ws:', 'wss:'].includes(forUrl.protocol)) {
```
nit, easier to follow imo.

Reply:
I think it is better to put the comment inside the block, because then it is clear whether the comment is about the "if" branch or "else" branch.